src/Test/Unit/WarmUp/UrlsPattern/CommandArgumentBuilderTest.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Test\Unit\WarmUp\UrlsPattern;
+
+use Codeception\PHPUnit\TestCase;
+use Magento\MagentoCloud\WarmUp\UrlsPattern\CommandArgumentBuilder;
+use PHPUnit\Framework\MockObject\MockObject;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @inheritDoc
+ */
+class CommandArgumentBuilderTest extends TestCase
+{
+    /**
+     * @var CommandArgumentBuilder
+     */
+    private $argumentBuilder;
+
+    /**
+     * @var LoggerInterface|MockObject
+     */
+    private $loggerMock;
+
+    /**
+     * @inheritDoc
+     */
+    protected function setUp()
+    {
+        $this->loggerMock = $this->getMockForAbstractClass(LoggerInterface::class);
+
+        $this->argumentBuilder = new CommandArgumentBuilder($this->loggerMock);
+    }
+
+    /**
+     * @param string $entity
+     * @param string $storeIds
+     * @param array $expected
+     * @dataProvider generateDataProvider
+     */
+    public function testGenerate(string $entity, string $storeIds, array $expected)
+    {
+        $this->loggerMock->expects($this->never())
```

Comment:
why do you add this check?

Reply:
To ensure that log message is writing only if `productIds == '*'`

Reply:
but in this method log is not used at all

Reply:
Yes, such check is redundant in this test method. Removed.

Reply:
you can. I guess it is a bug. not sure that someone expect this